src/foam/u2/Element.js

```diff
@@ -471,20 +478,28 @@ foam.CLASS({
   properties: [
     {
       class: 'Function',
-      name: 'addRow'
+      name: 'addRow',
+      documentation: `Takes a object from the the DAO and returns an Element.`
     },
     {
       class: 'Function',
-      name: 'cleanup'
+      name: 'cleanup',
+      documentation: `Should clean up the elements produced by addRow.`
     },
     'dao',
     {
       class: 'Int',
-      name: 'batch'
+      name: 'batch',
+      documentation: `Used to check whether a paint should be performed or not.`
     }
   ],
 
   methods: [
+    function init() {
+      // Repaint on every change to the DAO.
+      this.dao.sub('on', 'reset', this.paint);
```

Comment:
I don't think this should be done here. I don't think it's the RenderSink's responsibility and that the listen should be done in the Element.select() method.